openqaoa/problems/problem.py

```diff
@@ -639,9 +639,15 @@ def random_instance(**kwargs):
 
         values = list(map(int, np.random.randint(1, n_items, size=n_items)))
         weights = list(map(int, np.random.randint(1, n_items, size=n_items)))
-        weight_capacity = np.random.randint(np.min(weights) * n_items, np.max(weights) * n_items)
-        penalty = 2 * np.max(values)
 
+        min_weights = np.min(weights)
+        max_weights = np.max(weights)
+        if min_weights * n_items + 1 < max_weights * n_items :
```

Comment:
Since the edge case is when `min_weights == max_weights` ,maybe it makes sense to check for that instead. I'm not sure if im missing something here.

Reply:
fixed